frame/evm/src/lib.rs

```diff
@@ -821,32 +821,14 @@ impl<T: Config> Pallet<T> {
 	/// Remove an account.
 	pub fn remove_account(address: &H160) {
 		if <AccountCodes<T>>::contains_key(address) {
-			// Remember to call `dec_sufficients` when clearing Suicided.
-			<Suicided<T>>::insert(address, ());
-
-			// In theory, we can always have pre-EIP161 contracts, so we
-			// make sure the account nonce is at least one.
 			let account_id = T::AddressMapping::into_account_id(*address);
-			frame_system::Pallet::<T>::inc_account_nonce(&account_id);
+			let _ = frame_system::Pallet::<T>::dec_sufficients(&account_id);
 		}
 
 		<AccountCodes<T>>::remove(address);
 		<AccountCodesMetadata<T>>::remove(address);
-
-		if T::SuicideQuickClearLimit::get() > 0 {
-			#[allow(deprecated)]
-			let res = <AccountStorages<T>>::remove_prefix(address, Some(T::SuicideQuickClearLimit::get()));
-
-			match res {
-				KillStorageResult::AllRemoved(_) => {
-					<Suicided<T>>::remove(address);
-
-					let account_id = T::AddressMapping::into_account_id(*address);
-					let _ = frame_system::Pallet::<T>::dec_sufficients(&account_id);
-				}
-				KillStorageResult::SomeRemaining(_) => (),
-			}
-		}
+		#[allow(deprecated)]
+		let _ = <AccountStorages<T>>::remove_prefix(address, None);
```

Comment:
Lets avoid using this deprecated method

```suggestion
		let _ = <AccountStorages<T>>::clear_prefix(address, u32::max_value(), None);
```